src/chronify/store.py

```diff
@@ -1,45 +1,188 @@
+
     def create_view_from_parquet(self, name: str, path: Path) -> None:
         """Create a view in the database from a Parquet file."""
         with self._engine.begin() as conn:
-            query = f"CREATE VIEW {name} AS SELECT * FROM read_parquet('{path}/**/*.parquet')"
+            if self._engine.name == "duckdb":
+                path_ = f"{path}/**/*.parquet" if path.is_dir() else path
+                query = f"CREATE VIEW {name} AS SELECT * FROM read_parquet('{path_}')"
+            else:
+                msg = f"create_view_from_parquet does not support engine={self._engine.name}"
+                raise NotImplementedError(msg)
             conn.execute(text(query))
+            conn.commit()
+        self.update_table_schema()
+
+    def ingest_from_csv(
+        self,
+        path: Path | str,
+        src_schema: CsvTableSchema,
+        dst_schema: TableSchema,
+    ) -> None:
+        """Ingest data into the table specifed by schema. If the table does not exist,
+        create it.
+        """
+        rel = read_csv(path, src_schema)
+        check_columns(rel.columns, src_schema.list_columns())
+        # TODO: doesn't do anything yet
+        check_schema_compatibility(src_schema, dst_schema)
+
+        if src_schema.pivoted_dimension_name is not None:
+            rel = ddbf.unpivot(
+                rel,
+                src_schema.value_columns,
+                src_schema.pivoted_dimension_name,
+                dst_schema.value_column,
+            )
+
+        if isinstance(src_schema.time_config, IndexTimeRange):
+            if isinstance(dst_schema.time_config, DatetimeRange):
+                rel = ddbf.add_datetime_column(
+                    rel=rel,
+                    start=dst_schema.time_config.start,
+                    resolution=dst_schema.time_config.resolution,
+                    length=dst_schema.time_config.length,
+                    time_array_id_columns=src_schema.time_array_id_columns,
+                    time_column=dst_schema.time_config.time_column,
+                    timestamps=src_schema.time_config.list_timestamps(),
+                )
+            else:
+                cls_name = dst_schema.time_config.__class__.__name__
+                msg = f"IndexTimeRange cannot be converted to {cls_name}"
+                raise NotImplementedError(msg)
+
+        if self.has_table(dst_schema.name):
+            table = Table(dst_schema.name, self._metadata)
+        else:
+            dtypes = [get_sqlalchemy_type_from_duckdb(x) for x in rel.dtypes]
+            columns = [Column(x, y) for x, y in zip(rel.columns, dtypes)]
+            table = Table(dst_schema.name, self._metadata, *columns)
+            table.create(self._engine)
+
+        with self._engine.begin() as conn:
+            write_database(rel.to_df(), conn, dst_schema)
+            conn.commit()
+        self.update_table_schema()
+
+    def read_query(self, query: Selectable | str, schema: TableSchema) -> pd.DataFrame:
```

Comment:
There is currently an unfortunate limitation here. You have to pass the schema because SQLite is forcing us to perform custom logic on the time columns. We don't know which columns are time columns yet. We need a second table to store that metadata. I propose that we defer that to a subsequent PR.